spec/grape_entity/entity_spec.rb

```diff
@@ -991,6 +996,110 @@ def embedded
           expect(presenter.serializable_hash).to eq(name: 'abc', embedded: { a: nil, b: { abc: 'def' } })
         end
       end
+
+      context '#attr_path' do
+        it 'for all kinds of attributes' do
+          module EntitySpec
+            class EmailEntity < Grape::Entity
+              expose(:email, as: :addr) { |_, o| o[:attr_path].join('/') }
+            end
+
+            class UserEntity < Grape::Entity
+              expose(:name, as: :full_name) { |_, o| o[:attr_path].join('/') }
+              expose :email, using: 'EntitySpec::EmailEntity'
+            end
+
+            class ExtraEntity < Grape::Entity
+              expose(:key) { |_, o| o[:attr_path].join('/') }
+              expose(:value) { |_, o| o[:attr_path].join('/') }
+            end
+
+            class NestedEntity < Grape::Entity
+              expose(:name) { |_, o| o[:attr_path].join('/') }
+              expose :data, using: 'EntitySpec::ExtraEntity'
+            end
+          end
+
+          fresh_class.class_eval do
+            expose(:id) { |_, o| o[:attr_path].join('/') }
+            expose(:foo, as: :bar) { |_, o| o[:attr_path].join('/') }
+            expose :title do
+              expose :full do
+                expose(:prefix, as: :pref) { |_, o| o[:attr_path].join('/') }
+                expose(:main) { |_, o| o[:attr_path].join('/') }
+              end
+            end
+            expose :friends, as: :social, using: 'EntitySpec::UserEntity'
+            expose :extra, using: 'EntitySpec::ExtraEntity'
+            expose :nested, using: 'EntitySpec::NestedEntity'
+          end
+
+          expect(subject.serializable_hash).to eq(
+            id: 'id',
+            bar: 'bar',
+            title: { full: { pref: 'title/full/pref', main: 'title/full/main' } },
+            social: [
+              { full_name: 'social/full_name', email: { addr: 'social/email/addr' } },
+              { full_name: 'social/full_name', email: { addr: 'social/email/addr' } }
+            ],
+            extra: { key: 'extra/key', value: 'extra/value' },
+            nested: [
+              { name: 'nested/name', data: { key: 'nested/data/key', value: 'nested/data/value' } },
+              { name: 'nested/name', data: { key: 'nested/data/key', value: 'nested/data/value' } }
+            ]
+          )
+        end
+
+        it 'allows customize path of an attribute' do
+          module EntitySpec
+            class CharacterEntity < Grape::Entity
+              expose(:key) { |_, o| o[:attr_path].join('/') }
+              expose(:value) { |_, o| o[:attr_path].join('/') }
+            end
+          end
+
+          fresh_class.class_eval do
+            expose :characteristics, using: EntitySpec::CharacterEntity
+
+            protected
+
+            def self.path_for(attribute)
+              attribute == :characteristics ? :character : super
```

Comment:
I don't like this idea. This thing should be implemented in exposure parameters. For example:

```ruby
expose :characteristics, using: EntitySpec::CharacterEntity, attr_path: proc { :character }
```
What if we have more attributes which path we need to override? We'll end up with conditional hell in `path_for`.